docs/assets/javascript/elements/uxdot-pattern.css

```diff
@@ -1,89 +1,96 @@
+@import url('/styles/rh-code-block-lightdom.css');
+
 :host {
+  container: host / inline-size;
   display: block;
-  container-name: host;
-  container-type: inline-size;
-  margin-block-end: var(--rh-space-2xl, 32px);
+  margin-block-start: var(--rh-space-2xl);
 }
 
 #container {
   display: grid;
-  grid-template-columns: 1fr;
-  grid-template-areas: 'controls' 'example' 'code';
-  gap: var(--rh-space-2xl, 32px) var(--rh-space-4xl, 64px);
-  padding: var(--rh-space-2xl, 32px);
-  border: var(--rh-border-width-sm, 1px) solid var(--rh-color-border-subtle);
-  border-radius: var(--rh-border-radius-default, 3px);
-  background-color: var(--rh-color-surface);
-  color: var(--rh-color-text-primary);
-  border-color: var(--rh-color-border-subtle);
+  grid-template-columns: auto auto;
+  grid-template-areas: 'heading' 'controls' 'content' 'code';
+  gap: var(--rh-space-2xl) var(--rh-space-4xl);
 
-  --rh-color-surface: inherit;
-}
+  @container host (width >= 530px) {
+    grid-template-areas:
+      'heading controls'
+      'content content'
+      'code code';
+  }
 
-#container h3,
-#container ::slotted(h2),
-#container ::slotted(h3),
-#container ::slotted(h4),
-#container ::slotted(h5),
-#container ::slotted(h6) {
-  margin-block: 0 var(--rh-space-lg, 16px) !important;
-  font-size: var(--rh-font-size-body-text-lg, 1.125rem) !important;
+  @container host (width >= 992px) {
+    grid-template-areas:
+      'heading heading'
+      'controls controls'
+      'example example'
+      'code code';
+  }
 }
 
-#color-picker {
-  grid-area: controls;
-  justify-self: flex-end;
+#heading {
+  grid-area: heading;
   display: flex;
   align-items: center;
-  gap: var(--rh-space-lg, 16px);
 }
 
-#example {
-  grid-area: example;
-}
+#description { grid-area: content; }
 
-:host([stacked]) #example ::slotted(*) {
-  max-width: 100%;
+h3,
+::slotted(h2),
+::slotted(h3),
+::slotted(h4),
+::slotted(h5),
+::slotted(h6) {
+  margin-block: 0 !important;
+  font-size: var(--rh-font-size-body-text-lg) !important;
 }
 
-#code {
-  grid-area: code;
-  display: flex;
-  flex-direction: column;
-  gap: var(--rh-space-lg, 16px);
-}
+#content {
+  display: block;
+  resize: vertical;
+  overflow: auto;
+  grid-area: content;
+  padding: var(--rh-space-4xl);
+  border: var(--rh-border-width-sm) solid var(--rh-color-border-subtle);
+  border-radius: var(--rh-border-radius-default);
+  background-color: var(--rh-color-surface);
+  color: var(--rh-color-text-primary);
+  border-color: var(--rh-color-border-subtle);
 
-#container.noColorPicker {
-  grid-template-areas: 'example' 'code';
+  @container host (width < 300px) {
+    padding: var(--rh-space-lg);
+  }
```

Comment:
I'm thinking here we'd want to base this off the more narrow container so maybe we need a wrapping div around the `rh-surface` called `#surface-container` and `{ container: surface / inline-size }` ?

Reply:
why do we need that? i think that would just add complexity. We only have one vertical layout, and all we need to do is adjust padding to suit smaller containers (IMO)

Reply:
[6a6eabc](https://github.com/RedHat-UX/red-hat-design-system/pull/1971/commits/6a6eabc1072356fb05337ad7441b314b7296934b)